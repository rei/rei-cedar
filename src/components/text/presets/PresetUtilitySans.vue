<script setup lang="ts">
import { useCssModule } from 'vue';
import CdrText from '../CdrText.vue'

const baseClass = 'cdr-utility-sans';
const style = useCssModule();
</script>

<template>
  <CdrText :class="style[baseClass]">
    <slot />
  </CdrText>
</template>

<style module lang="scss">
@import "@rei/cdr-tokens/dist/rei-dot-com/scss/cdr-tokens.scss";
@import "../../../../public/cdr-fluid-type.css";

.cdr-utility-sans {
  @include cdr-text-utility-sans-500;
  font-size: var(--cdr-type-scale-0);
  line-height: calc(var(--cdr-type-scale-1) * var(--cdr-utility-sans-line-height-scale));
}
</style>  