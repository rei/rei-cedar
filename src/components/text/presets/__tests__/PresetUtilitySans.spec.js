import { mount } from '../../../../../test/vue-jest-style-workaround.js';
import PresetUtilitySans from '../PresetUtilitySans.vue';


describe('PresetUtilitySans', () => {
  describe('default configuration', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(PresetUtilitySans, {
        slots: {
          default: 'paragraph'
        }
      });
    });
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('passing a scale', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(PresetUtilitySans, {
        props: { scale: '-1' },
        slots: {
          default: 'paragraph'
        }
      });
    });
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('setting to strong variant', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(PresetUtilitySans, {
        props: { strong: true },
        slots: {
          default: 'paragraph'
        }
      });
    });
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

});
