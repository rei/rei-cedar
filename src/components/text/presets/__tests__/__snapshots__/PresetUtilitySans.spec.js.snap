// Vitest Snapshot v1

exports[`PresetUtilitySans > default configuration > renders correctly 1`] = `
<p
  class="cdr-text cdr-utility-sans"
  style="--cdr-utility-sans-font-size: var(--cdr-type-scale-1); --cdr-utility-sans-line-height: calc(var(--cdr-type-scale-1) * var(--cdr-utility-sans-line-height-ratio)); --cdr-utility-sans-font-weight: 400;"
>
  <!-- @slot CdrText content -->
  
  
  paragraph
  
  
</p>
`;

exports[`PresetUtilitySans > passing a scale > renders correctly 1`] = `
<p
  class="cdr-text cdr-utility-sans"
  style="--cdr-utility-sans-font-size: var(--cdr-type-scale--1); --cdr-utility-sans-line-height: calc(var(--cdr-type-scale--1) * var(--cdr-utility-sans-line-height-ratio)); --cdr-utility-sans-font-weight: 400;"
>
  <!-- @slot CdrText content -->
  
  
  paragraph
  
  
</p>
`;

exports[`PresetUtilitySans > setting to strong variant > renders correctly 1`] = `
<p
  class="cdr-text cdr-utility-sans"
  style="--cdr-utility-sans-font-size: var(--cdr-type-scale-1); --cdr-utility-sans-line-height: calc(var(--cdr-type-scale-1) * var(--cdr-utility-sans-line-height-ratio)); --cdr-utility-sans-font-weight: 600;"
>
  <!-- @slot CdrText content -->
  
  
  paragraph
  
  
</p>
`;
